components/api-controller/cmd/controller/main.go

```diff
@@ -144,6 +146,10 @@ func getCORSConfig() istioNetworkingV1.CORSConfig {
 	}
 }
 
+func readBlacklistedServices(list string) []string {
+	return removeEmptyStrings(strings.Split(list, ","))
```

Comment:
this line: removeEmptyStrings(strings.Split(list, ",")) can be re-used with three other places.